go/vt/tabletserver/twopc.go

```diff
@@ -470,11 +470,12 @@ func (tpc *TwoPC) ReadAllTransactions(ctx context.Context) ([]*DistributedTx, er
 		dtid := row[0].String()
 		if curTx == nil || dtid != curTx.Dtid {
 			// Initialize the new element.
-			v, _ := strconv.ParseInt(row[2].String(), 10, 64)
+			tm, _ := strconv.ParseInt(row[2].String(), 10, 64)
+			st, _ := strconv.ParseInt(row[1].String(), 10, 64)
 			curTx = &DistributedTx{
 				Dtid:    dtid,
-				State:   row[1].String(),
-				Created: time.Unix(0, v),
+				State:   querypb.TransactionState(st).String(),
```

Comment:
This could have an out-of-range value, let's catch that. I think proto enums have a map with all possible values in it.